src/context/WalletProvider/MobileWallet/components/MobileLoad.tsx

```diff
@@ -0,0 +1,196 @@
+import { Row } from 'components/Row/Row'
+import { RawText, Text } from 'components/Text'
+import { WalletActions } from 'context/WalletProvider/actions'
+import { KeyManager } from 'context/WalletProvider/KeyManager'
+import {
+  setLocalNativeWalletName,
+  setLocalWalletTypeAndDeviceId,
+} from 'context/WalletProvider/local-wallet'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+import { MobileConfig, mobileLogger } from '../config'
+import { deleteWallet, getWallet, listWallets } from '../mobileMessageHandlers'
+import { RevocableWallet } from '../RevocableWallet'
+
+const moduleLogger = mobileLogger.child({
+  namespace: ['WalletProvider', 'MobileWallet', 'components', 'MobileLoad'],
+})
+
+export const MobileLoad = ({ history }: RouteComponentProps) => {
+  const { state, dispatch } = useWallet()
+  const [error, setError] = useState<string | null>(null)
+  const [wallets, setWallets] = useState<RevocableWallet[]>([])
+  const translate = useTranslate()
+
+  useEffect(() => {
+    ;(async () => {
+      if (!wallets.length) {
+        try {
+          const vaults = await listWallets()
+          moduleLogger.trace({ vaults }, 'Found wallets')
+          if (!vaults.length) {
+            return setError('walletProvider.shapeShift.load.error.noWallet')
+          }
+
+          setWallets(vaults)
+        } catch (e) {
+          mobileLogger.error(e, 'Error reading list of wallets')
+          setWallets([])
+        }
+      }
+    })()
+  }, [wallets])
+
+  const handleWalletSelect = async (item: RevocableWallet) => {
+    const adapter = state.adapters?.get(KeyManager.Native)
+    const deviceId = item?.id
+    if (adapter && deviceId) {
+      const { name, icon } = MobileConfig
+      try {
+        const revoker = await getWallet(deviceId)
+        if (!revoker?.mnemonic) throw new Error(`Mobile wallet not found: ${deviceId}`)
+
+        const wallet = await adapter.pairDevice(revoker.id)
+        await wallet.loadDevice({ mnemonic: revoker.mnemonic })
+        if (!(await wallet.isInitialized())) {
+          await wallet.initialize()
+        }
+        dispatch({
+          type: WalletActions.SET_WALLET,
+          payload: { wallet, name, icon, deviceId, meta: { label: item.label } },
+        })
+        dispatch({ type: WalletActions.SET_IS_CONNECTED, payload: true })
+        dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: false })
+
+        setLocalWalletTypeAndDeviceId(KeyManager.Mobile, deviceId)
+        setLocalNativeWalletName(item?.label ?? 'label')
+      } catch (e) {
+        mobileLogger.error(e, { deviceId }, 'Error loading a wallet')
+        setError('walletProvider.shapeShift.load.error.pair')
+      }
+    } else {
+      mobileLogger.warn({ deviceId }, 'Missing adapter or device ID')
+      setError('walletProvider.shapeShift.load.error.pair')
+    }
+  }
+
+  const handleDelete = async (wallet: RevocableWallet) => {
+    const result = window.confirm(
```

Comment:
@DiggyDiggy2 this will pop up a browser-native confirm modal - we may want a nicer confirm handle delete modal before shipping to users?

Reply:
Sent to product so we don't forget about this one, resolving for now